src/Fantomas/CodePrinter.fs

```diff
@@ -1660,10 +1660,10 @@ and genClause astContext hasBar (Clause(p, e, eo) as node) =
         else
             preserveBreakNln astContext e ctx
 
+    let pat = genPat astContext p
+    let body = optPre (!- " when ") sepNone eo (genExpr astContext) +> sepArrow +> clauseBody e
     genTriviaBeforeClausePipe p.Range +>
-    ifElse hasBar sepBar sepNone +> genPat astContext p
-    +> optPre (!- " when ") sepNone eo (genExpr astContext) +> sepArrow
-    +> clauseBody e
+    ifElse hasBar (sepBar +> atCurrentColumnWithPrepend pat body) (pat +> body)
```

Comment:
I'm a bit lost what `atCurrentColumnWithPrepend ` does at this point.
In the example of 
```fsharp
| Some y ->
    let z = 1
    Some(y + z)
```
It will print `Some y` first? And what happens then? Because the arrow `->` is still right after `pat` so how does the `atIndentLevel` work then?

Reply:
Its about that we store current column (at start of `Some`), then do `prependF` without setting indentation column, then set indentation column to stored value and then do rest.

This is needed for multiline pattern like
```
 | P1 y 
 | P2 y ->
    let z = 1
    Some(y + z)
```

I am all in for suggestion for better name of better comment for `atCurrentColumnWithPrepend` :)